### ISSUE TITLE
Support loopback.getCurrentContext in the browser

### ISSUE THREAD

**Issue by BerkeleyTrue** [strongloop/loopback#1020]
This would help with things like storing user sessions and access token in localStorage while using an existing api.

``` js
// server

var ctx = loopback.getCurrentContext();

ctx.set('currentUser', user); // Uses continuous-local-storage and domains

// client

var  ctx = loopback.getCurrentContext();

ctx.set('currentUser', user); // Uses localStorage 

```

**Comment by ritch** [comment]
+1

Although I am a bit worried about building on top of `getCurrentContext()`. It is very slow...

**Comment by bajtos** [comment]
I don't like this idea:
 - LocalStorage is persistent, the "current context" is not meant to be persisted.
 - When there are two methods called in parallel, each invocation has it's own "current context". LocalStorage-backed "current context" would be shared by all invocations.

We should find a different way how to provide "current context" functionality in a browser.

**Comment by bajtos** [comment]
We should get this for free once https://github.com/othiym23/async-listener/issues/14 is implemented and released.

**Comment by pthieu** [comment]
Closing in favour of https://github.com/othiym23/async-listener/issues/14. Please re-open if this is still required, @BerkeleyTrue.
